### PR TITLE
feat: Add support for private files to file module

### PR DESCRIPTION
**Pull request by sradevski**
CLOSES CC-213

### CODE REVIEW

File: packages/modules/providers/file-s3/integration-tests/__tests__/services.spec.ts [hunk 1]
```diff
@@ -35,45 +35,50 @@ describe.skip("S3 File Plugin", () => {
...
+      expect(signedUrlFile.toString("binary")).toEqual(fixtureAsBinary)
 
-    await s3Service.delete({ fileKey: resp.key })
+      await s3Service.delete({ fileKey: resp.key })
 
-    // TODO: Currently the presignedURL will be returned even if the file doesn't exist. Should we check for existence first?
-    const deletedFileUrl = await s3Service.getPresignedDownloadUrl({
-      fileKey: resp.key,
-    })
+      // TODO: Currently the presignedURL will be returned even if the file doesn't exist. Should we check for existence first?
```
Comment (olivermrbl) [inline packages/modules/providers/file-s3/integration-tests/__tests__/services.spec.ts:72]: It sounds reasonable to check for existence in `getPresignedDownloadUrl` and throw `File not found` or similar, if that is the case. Wdyt?

**Review by olivermrbl — APPROVED** [review]
LGTM

File: packages/modules/providers/file-s3/integration-tests/__tests__/services.spec.ts (hunk 1 above)
Comment (sradevski) [reply packages/modules/providers/file-s3/integration-tests/__tests__/services.spec.ts:72]: @olivermrbl Yeah I think it makes sense, the only downside is that it requires one more request, so I was a bit unsure if we want to make the 99% of the requests slower for better error message.

Now that I think about it though, presigned URLs are only valid for private files, which will be a lot less common, so maybe we can add the check. WDYT?

Btw I'll just merge this PR as it is unrelated, and depending on what we decide I'll open a new one.

Comment (olivermrbl) [reply packages/modules/providers/file-s3/integration-tests/__tests__/services.spec.ts:72]: Yeah, sounds good to me. No need to rush with the follow-up PR though. Can be tackled later.

Comment (sradevski) [reply packages/modules/providers/file-s3/integration-tests/__tests__/services.spec.ts:72]: I'll just open a ticket for now

**Pull request merged**